mmtk/src/api.rs

```diff
@@ -106,17 +104,25 @@ pub extern "C" fn alloc(
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub extern "C" fn post_alloc(
     mutator: *mut Mutator<JikesRVM>,
-    refer: ObjectReference,
-    _type_refer: ObjectReference,
+    refer: JikesObj,
+    _type_refer: JikesObj,
     bytes: usize,
     allocator: AllocationSemantics,
 ) {
-    memory_manager::post_alloc::<JikesRVM>(unsafe { &mut *mutator }, refer, bytes, allocator)
+    debug_assert!(!refer.is_null());
```

Comment:
The null check seems unnecessary. There is always `try_into/from` afterwards, which will fail if you have a null reference.

Reply:
I removed all `debug_assert!(!jikes_obj.is_null());` from API functions if the `JikesObj` parameter is converted to `ObjectReference` using `try_from` or `try_into`.